src/containers/BrowseMoviesContainer.js

```diff
@@ -64,13 +70,44 @@ class BrowseMoviesContainer extends Component {
     }
   }
 
+  loadMoreAndAppend = async () => {
+    const { filter, id } = this.props.match.params;
+    try {
+      let resp;
+      if (filter !== "genre") {
+        resp = await getMoviesFromType(filter, this.state.currentPage + 1);
+      } else {
+        resp = await getGenreMovies(id, this.state.currentPage + 1);
+      }
+      const index = this.state.movies.concat(resp.results);
+      const resArr = [];
+      index.filter((item) => {
```

Comment:
What's happening in this filter block?

Reply:
First i concat the old movies with the new movies from the API and then the filter block removes duplicates to avoid duplicate movies in the grid view and multiple key errors in console

Reply:
Ok, but since `filter` returns an array, I think it will be more "correct" to use a `foreach` statement:

```js
index.forEach((item) => {
  const i = resArr.findIndex(x => x.id === item.id);
  if (i <= -1) {
    resArr.push(item);
  }
}
```

Reply:
You were surprised there were foreaches in javascript and u wanted to know the difference. Well here you have a use case for it :+1:

Reply:
Also, it's not obvious what it does, so maybe add a short comment saying that it removes duplicates